Source/WebCore/css/CSSFontFaceSet.cpp

```diff
@@ -117,19 +117,21 @@ void CSSFontFaceSet::ensureLocalFontFacesForFamilyRegistered(const AtomString& f
     if (m_locallyInstalledFacesLookupTable.contains(familyName))
         return;
 
-    if (!m_owningFontSelector->scriptExecutionContext())
+    Ref protectedOwningFontSelector { *m_owningFontSelector };
+    if (!protectedOwningFontSelector->scriptExecutionContext())
         return;
-    auto allowUserInstalledFonts = m_owningFontSelector->scriptExecutionContext()->settingsValues().shouldAllowUserInstalledFonts ? AllowUserInstalledFonts::Yes : AllowUserInstalledFonts::No;
+    Ref scriptExecutionContext { *protectedOwningFontSelector->scriptExecutionContext() };
```

Comment:
We should probably add `CSSFontSelector::protectedScriptExecutionContext()` and use it here.